torch/distributed/_composable/fsdp/_fsdp_param_group.py

```diff
@@ -275,6 +280,9 @@ def pre_backward(self, forward_grad_fns: Tuple[Any, ...], *unused: Any):
     def post_backward(self, *unused: Any):
         self._training_state = TrainingState.POST_BACKWARD
         with torch.profiler.record_function("FSDP::post_backward_reshard"):
+            if not self.reduce_scatter_grads:
```

Comment:
`if not self.reduce_scatter_grads and self.all_reduce_grads` we should also skip reshard?

Reply:
We disallow the case of no reduce-scatter but all-reduce. For HSDP, we either do not gradient sync (no reduce-scatter, no all-reduce) or just reduce-scatter (reduce-scatter, no all-reduce). If `not self.reduce_scatter_grads`, then that implies `not self.all_reduce_grads`.

Reply:
so this check would be set whenever we initialize the HSDP mode? (i.e. `self.reduce_scatter_grads==self.all_reduce_grads`)

Reply:
Yes. It would be an invariant that we preserve.